runelite-client/src/main/java/net/runelite/client/plugins/grounditems/GroundItemsPlugin.java

```diff
@@ -495,31 +495,18 @@ public void onMenuEntryAdded(MenuEntryAdded event)
 			MenuEntry[] menuEntries = client.getMenuEntries();
 			MenuEntry lastEntry = menuEntries[menuEntries.length - 1];
 
-			int quantity = 1;
-			Node current = itemLayer.getBottom();
-
-			while (current instanceof TileItem)
-			{
-				TileItem item = (TileItem) current;
-				if (item.getId() == itemId)
-				{
-					quantity = item.getQuantity();
-				}
-				current = current.getNext();
-			}
+			GroundItem.GroundItemKey groundItemKey = new GroundItem.GroundItemKey(itemId, tile.getWorldLocation());
+			GroundItem groundItem = collectedGroundItems.get(groundItemKey);
+			int quantity = groundItem.getQuantity();
 
-			final ItemComposition itemComposition = itemManager.getItemComposition(itemId);
-			final int realItemId = itemComposition.getNote() != -1 ? itemComposition.getLinkedNoteId() : itemComposition.getId();
-			final int itemPrice = itemManager.getItemPrice(realItemId);
-			final int price = itemPrice <= 0 ? itemComposition.getPrice() : itemPrice;
-			final int haPrice = Math.round(itemComposition.getPrice() * Constants.HIGH_ALCHEMY_MULTIPLIER) * quantity;
-			final int gePrice = quantity * price;
-			final Color hidden = getHidden(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice, itemComposition.isTradeable());
-			final Color highlighted = getHighlighted(new NamedQuantity(itemComposition.getName(), quantity), gePrice, haPrice);
+			final int gePrice = groundItem.getGePrice();
+			final int haPrice = groundItem.getHaPrice();
+			final Color hidden = getHidden(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice, groundItem.isTradeable());
+			final Color highlighted = getHighlighted(new NamedQuantity(groundItem.getName(), quantity), gePrice, haPrice);
 			final Color color = getItemColor(highlighted, hidden);
 			final boolean canBeRecolored = highlighted != null || (hidden != null && config.recolorMenuHiddenItems());
 
-			if (color != null && canBeRecolored && !color.equals(config.defaultColor()))
+			if (color != null && canBeRecolored)
```

Comment:
Why is this being removed? It would cause the plugin to apply the default overlay color to the menus which it currently doesn't do. It seems unrelated to this bug at least. I think the reason this exists is because the overlay needs a color but the menus already have their own default color which is white.

Reply:
If I understood your comment correctly, the default color should only be used for the overlay and not for the menu entry? I recommend changing it so both match the configured colors(highlight, default, hidden). Otherwise I'd recommend changing the description of the color so that's it's clear that the default color doesn't change the default color of menu entries.

Reply:
This would cause a rather user visible behavior change, which generally if we do those (especially without a config option toggle) it upsets people. At the very least it is not related to the two bugs this is trying to fix though.